### DISCUSSION TITLE
[과목 1] 제 2장 데이터 모델과 SQL

### DISCUSSION THREAD

**Discussion by beolggoolosori** [flataex/sqlp-study#7] [Custom 연습문제] [answered]
>1번 문제

아래의 테이블을 제 3 정규형으로 정규화하시오.

id | customer_name | product_name | product_price | order_date
-- | -- | -- | -- | --
1 | 김종훈 | 초코송이 | 2200 | 2023-01-01
2 | 한백규 | 고래밥 | 2500 | 2023-01-02
3 | 권도영 | 포카칩 | 2000 | 2023-01-03
4 | 박지은 | 추파춥스 | 300 | 2023-01-04

<br/><br/>
>2번 문제

first_name | last_name | birthdate | city | email
-- | -- | -- | -- | --
도영 | 권 | 1980-01-01 | 서울 | dykwon@flataex.com
지은 | 박 | 1985-02-02 | 부산 | jepark@flataex.com
종훈 | 김 | 1995-03-03 | 대구 | jhkim@flataex.com

 다음 중 고객 정보 테이블에서 완벽한 고유성을 가질 수 있는 열은 무엇입니까?

① first_name + last_name
② birthdate + city
③ birthdate + postal_code
④ email

**Comment by white-gyu** [comment, marked as answer]
>1번 문제

아래의 테이블을 제 3 정규형으로 정규화하시오.

id | customer_name | product_name | product_price | order_date
-- | -- | -- | -- | --
1 | 김종훈 | 초코송이 | 2200 | 2023-01-01
2 | 한백규 | 고래밥 | 2500 | 2023-01-02
3 | 권도영 | 포카칩 | 2000 | 2023-01-03
4 | 박지은 | 추파춥스 | 300 | 2023-01-04

<br>
<br>

<table>
<tr>
<td align="center" colspan="2">customer 테이블</td>
</tr>
<tr>
<td align="center">id</td><td align="center">name</td>
</tr>
<tr>
<td align="center">1</td><td align="center">김종훈</td>
</tr>
<tr>
<td align="center">2</td><td align="center">한백규</td>
</tr>
<tr>
<td align="center">3</td><td align="center">권도영</td>
</tr>
<tr>
<td align="center">4</td><td align="center">박지은</td>
</tr>
</table>

<table>
<tr>
<td align="center" colspan="3">product 테이블</td>
</tr>
<tr>
<td align="center">id</td><td align="center">name</td><td align="center">price</td>
</tr>
<tr>
<td align="center">1</td><td align="center">초코송이</td><td align="center">2200</td>
</tr>
<tr>
<td align="center">2</td><td align="center">고래밥</td><td align="center">2500</td>
</tr>
<tr>
<td align="center">3</td><td align="center">포카칩</td><td align="center">2000</td>
</tr>
<tr>
<td align="center">4</td><td align="center">추파춥스</td><td align="center">300</td>
</tr>
</table>

<table>
<tr>
<td align="center" colspan="4">order 테이블</td>
</tr>
<tr>
<td align="center">id</td><td align="center">customer_id</td><td align="center">product_id</td><td align="center">order_date</td>
</tr>
<tr>
<td align="center">1</td><td align="center">1</td><td align="center">1</td><td align="center">2023-01-01</td>
</tr>
<tr>
<td align="center">2</td><td align="center">2</td><td align="center">2</td><td align="center">2023-01-02</td>
</tr>
<tr>
<td align="center">3</td><td align="center">3</td><td align="center">3</td><td align="center">2023-01-03</td>
</tr>
<tr>
<td align="center">4</td><td align="center">4</td><td align="center">4</td><td align="center">2023-01-04</td>
</tr>
</table>


---

>2번 문제

first_name | last_name | birthdate | city | email
-- | -- | -- | -- | --
도영 | 권 | 1980-01-01 | 서울 | dykwon@flataex.com
지은 | 박 | 1985-02-02 | 부산 | jepark@flataex.com
종훈 | 김 | 1995-03-03 | 대구 | jhkim@flataex.com


 다음 중 고객 정보 테이블에서 완벽한 고유성을 가질 수 있는 열은 무엇입니까?

① first_name + last_name
② birthdate + city
③ birthdate + postal_code
④ email

> ④

**Reply by beolggoolosori** [reply]
정확합니다.
